src/msix/PAL/FileSystem/POSIX/DirectoryObject.cpp

```diff
@@ -9,9 +9,44 @@
 #include <sys/stat.h>
 #include <errno.h>
 #include <fts.h>
+#include <dirent.h>
+#include <map>
 
 namespace MSIX {
 
+    template<class Lambda>
+    void WalkDirectory(const std::string& root, Lambda& visitor)
+    {
+        static std::string dot(".");
+        static std::string dotdot("..");
+
+        std::unique_ptr<DIR, decltype(&closedir)> dir(opendir(root.c_str()), closedir);
+        ThrowErrorIf(Error::FileNotFound, dir.get() == nullptr, "Invalid directory");
+        struct dirent* dp;
+        while((dp = readdir(dir.get())) != nullptr)
+        {
+            std::string fileName = std::string(dp->d_name);
+            std::string child = root + "/" + fileName;
+            if (dp->d_type == DT_DIR)
+            {
+                if ((fileName != dot) && (fileName != dotdot))
+                {
+                    WalkDirectory(child, visitor);

```

Comment:
>WalkDirectory [](start = 20, length = 13)

Potential recursion problems, especially if we don't handle potential junction loops.

Reply:
But not handling junction loops will break any impl

---
In reply to: [276761084](https://github.com/Microsoft/msix-packaging/pull/93#discussion_r276761084) [](ancestors = 276761084)

Reply:
I'll add a bug to this and fix it later

---
In reply to: [276761259](https://github.com/Microsoft/msix-packaging/pull/93#discussion_r276761259) [](ancestors = 276761259,276761084)